src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+        base_python = base_python[0]
+        if not base_python.endswith(version_nodot):
+            raise RuntimeError(
+                "Python version mismatch. "
+                f"Current version: {sys.version_info[:2]}, "
+                f"Requested environment: {base_python}"
+            )
+
+        return PythonInfo(
+            implementation=sys.implementation,
+            version_info=sys.version_info,
+            version=sys.version,
+            is_64=(platform.architecture()[0] == "64bit"),
+            platform=platform.platform(),
+            extra={"executable": Path(sys.executable)},
+        )
+
+    def create_python_env(self):
+        return None
+
+    def env_bin_dir(self):
+        return Path(sys.prefix) / "bin"
+
+    def env_python(self):
+        return sys.executable
+
+    def env_site_package_dir(self):
+        return Path(sys.prefix) / "lib"
+
+    @property
+    def installer(self):
+        if self._installer is None:
+            self._installer = DummyInstaller()
+        return self._installer
+
+    def prepend_env_var_path(self):
+        return [self.env_bin_dir()]
+
+    @property
+    def runs_on_platform(self):
+        return sys.platform
+
+
+class CurrentEnvRunExecutor(Execute):
+    def build_instance(
+        self,
+        request,
+        options,
+        out,
+        err,
+    ):
+        # Disable check for the external commands,
+        # all of them are external for us.
+        request.allow = None
+        return LocalSubProcessExecuteInstance(request, options, out, err)
+
+
+class DummyInstaller:
+    def install(self, *args):
+        return
+
+
+class PrintEnv(CurrentEnv):
+    def __init__(self, create_args):
+        super().__init__(create_args)
+
+        # As soon as this environment has enough info to do its job,
+        # do it and nothing more.
+
+        if self.options.print_deps_to:
+            print(
+                *self.core["requires"],
+                *self.conf["deps"].lines(),
+                sep="\n",
+                file=self.options.print_deps_to,
+            )
+            self.options.print_deps_to.flush()
+
+        if self.options.print_extras_to:
+            if "extras" not in self.conf:
+                # Unfortunately, if there is skipsdist/no_package or skip_install
+                # in the config, this section is not parsed at all so we have to
+                # do it here manually to be able to read its content.
+                self.conf.add_config(
+                    keys=["extras"],
+                    of_type=Set[str],
+                    default=set(),
+                    desc="extras to install of the target package",
+                )
+            print(
+                *self.conf["extras"],
+                sep="\n",
+                file=self.options.print_extras_to,
+            )
+            self.options.print_extras_to.flush()
+
+        # We are done
+        sys.exit(0)
```

Comment:
This is incorrect. This should be a new sub-command. We don't support calling sys.exit, and this will likely be a no-op when called with -p.

Reply:
If you don't want to make it a subcommand a better alternative would be to set commands to none and then do this within https://tox.wiki/en/rewrite/plugins.html#tox.plugin.spec.tox_before_run_commands

Reply:
I'll probably do both things. We want to keep the stable API so I clear the list of commands and do this in the `tox_before_run_commands` and I'll also prepare new subcommand. We can then deprecate the old CLI options and stay with subcommands only.